--- ./utils/debug.h.orig	2012-03-26 07:24:57.000000000 -0600
+++ ./utils/debug.h	2012-05-03 13:48:30.764978078 -0600
@@ -108,7 +108,7 @@ throw FFPACK::Failure (__func__, __FILE_
 
 
 namespace FFPACK {
-
+namespace {
 
 	/*!  A precondtion failed.
 	 * @ingroup util
@@ -215,6 +215,7 @@ namespace FFPACK {
 #endif
 
 	std::ostream *Failure::_errorStream;
+}
 } // FFPACK
 
 #endif // __FFLASFFPACK_util_debug_H
