--- ./utils/debug.h.orig	2011-10-09 11:58:19.000000000 -0600
+++ ./utils/debug.h	2012-05-03 13:48:30.764978078 -0600
@@ -107,7 +107,7 @@ throw FFPACK::Failure (__func__, __FILE_
 
 
 namespace FFPACK {
-
+namespace {
 
 	/*!  A precondtion failed.
 	 * @ingroup util
@@ -214,6 +214,7 @@ namespace FFPACK {
 #endif
 
 	std::ostream *Failure::_errorStream;
+}
 } // FFPACK
 
 #endif // __FFLASFFPACK_util_debug_H
